connectors/src/connectors/microsoft/lib/graph_api.ts

```diff
@@ -60,6 +60,37 @@ export async function getFolders(
   return res.filter((item) => item.folder);
 }
 
+export async function getWorksheets(
+  client: Client,
+  internalId: string
+): Promise<MicrosoftGraph.WorkbookWorksheet[]> {
+  const { nodeType, itemApiPath } = microsoftNodeDataFromInternalId(internalId);
+
+  if (nodeType !== "file") {
+    throw new Error(
+      `Invalid node type: ${nodeType} for getWorksheets, expected file`
+    );
+  }
+
+  const res = await client.api(`${itemApiPath}/workbook/worksheets`).get();
+  return res.value;
+}
+
+export async function getWorksheetContent(
+  client: Client,
+  internalId: string
+): Promise<MicrosoftGraph.WorkbookRange> {
+  const { nodeType, itemApiPath } = microsoftNodeDataFromInternalId(internalId);
+
+  if (nodeType !== "worksheet") {
+    throw new Error(
```

Comment:
We should not throw directly!

Reply:
In that case it's fine I think because it's an unexpected, supposedly unreachable part of the code
Just like we throw e.g. on "connector not found" when we do a sync